src/vs/platform/environment/node/stdin.ts

```diff
@@ -46,13 +46,21 @@ export async function readFromStdin(targetPath: string, verbose: boolean): Promi
 
 	let encoding = await resolveTerminalEncoding(verbose);
 
-	const iconv = await import('iconv-lite');
+	const iconv = await import('iconv-lite-umd');
 	if (!iconv.encodingExists(encoding)) {
 		console.log(`Unsupported terminal encoding: ${encoding}, falling back to UTF-8.`);
 		encoding = 'utf8';
 	}
 
 	// Pipe into tmp file using terminals encoding
-	const converterStream = iconv.decodeStream(encoding);
-	process.stdin.pipe(converterStream).pipe(stdinFileStream);
+	const decoder = iconv.getDecoder(encoding);
+	process.stdin.on('error', stdinFileStream.destroy);
+	process.stdin.on('data', (chunk) => {
+		stdinFileStream.write(decoder.write(chunk));
+	});
+	process.stdin.on('end', () => {
+		stdinFileStream.write(decoder.end());
+		stdinFileStream.end();
+	});
+	process.stdin.on('close', stdinFileStream.close);
```

Comment:
@bpasero I am not sure how correct is what I am trying to achieve here. Also I am not entirely sure what would be the best way to test that everything is working as expected.

Alternatively we can change `iconv-lite-umd` to expose streaming interface and keep code here as it was.

Can you help me here to figure out which route to take?

Reply:
@gyzerok a way to test this is to run `ps aux | grep code | ./scripts/code-cli.sh -` which will pipe something into VSCode. The idea here is that if the terminal is using a non-utf8 encoding, we need to convert it. 

I think your code is doing OK, I looked into that earlier. Any objections to that path?

Reply:
@bpasero yeah, seems to work fine then. I prefer this way better because if we don't expose streaming API from `iconv-lite-umd` there is not chance that someone will accidentally use it in the wrong environment.

Can you restart the failed job? It looks like it was just some random failure. Otherwise it's ready then.

Reply:
Actually nvm with the job. I'll squash commits for cleaner history.